app/api/chat/engine/generate.ts

```diff
@@ -27,9 +27,12 @@ async function generateDatasource() {
   console.log(`Generating storage context for datasource '${datasource}'...`);
   // Split documents, create embeddings and store them in the storage context
   const ms = await getRuntime(async () => {
-    const storageContext = await storageContextFromDefaults({
-      persistDir: `${STORAGE_CACHE_DIR}/${datasource}`,
+    const milvusClient = getMilvusClient();
+    const vectorStore = new MilvusVectorStore({
+      milvusClient,
+      collection: datasource,
     });
+    const storageContext = await storageContextFromDefaults({ vectorStore });
```

Comment:
Try using the shared function getIndex here and use is the index directly in fromDocuments below (without storage context)